integration_tests/abstract_account_test.go

```diff
@@ -84,39 +94,46 @@ func TestXionAbstractAccount(t *testing.T) {
 	codeID, err := xion.StoreContract(ctx, xionUser.FormattedAddress(), path.Join(fp, "integration_tests", "testdata", "contracts", "account_updatable-aarch64.wasm"))
 	require.NoError(t, err)
 
+	// retrieve the hash
+	codeResp, err := ExecQuery(t, ctx, xion.FullNodes[0],
+		"wasm", "code-info", codeID)
+	require.NoError(t, err)
+	t.Logf("code response: %s", codeResp)
+
 	depositedFunds := fmt.Sprintf("%d%s", 100000, xion.Config().Denom)
 
-	// Register Abstract Account Using Public Key
 	registeredTxHash, err := ExecTx(t, ctx, xion.FullNodes[0],
 		xionUser.KeyName(),
-		"abstract-account", "register",
+		"xion", "register",
 		codeID,
-		fmt.Sprintf(`{"pubkey": "%s"}`, account["key"]),
+		xionUser.KeyName(),
 		"--funds", depositedFunds,
-		"--salt", "foo",
+		"--authenticator", "Secp256K1",
+		"--salt", "0",
+		"--authenticator-id", "0",
 		"--chain-id", xion.Config().ChainID,
 	)
+	require.NoError(t, err)
 
 	txDetails, err := ExecQuery(t, ctx, xion.FullNodes[0], "tx", registeredTxHash)
 	require.NoError(t, err)
+	t.Logf("TxDetails: %s", txDetails)
 	aaContractAddr := GetAAContractAddress(t, txDetails)
 
 	contractBalance, err := xion.GetBalance(ctx, aaContractAddr, xion.Config().Denom)
 	require.NoError(t, err)
 	require.Equal(t, uint64(100000), uint64(contractBalance))
 
-	/*
-			NOTE: Ideally we would use this metod, however the QueryContract formats the string making it harder to predict.
-		var ContractResponse interface{}
-			require.NoError(t, xion.QueryContract(ctx, aaContractAddr, fmt.Sprintf(`{"pubkey":{}}`), ContractResponse))
-	*/
-
-	contractState, err := ExecQuery(t, ctx, xion.FullNodes[0], "wasm", "contract-state", "smart", aaContractAddr, fmt.Sprintf(`{"pubkey":{}}`))
+	contractState, err := ExecQuery(t, ctx, xion.FullNodes[0], "wasm", "contract-state", "smart", aaContractAddr, fmt.Sprintf(`{"authenticator_by_i_d":{ "id": 0 }}`))
```

Comment:
Does this work with "authenticator_by_i_d" vs "authenticator_by_id"

Reply:
The contract is expecting the key like this `authenticator_by_i_d` becuase the type in the contract is `AuthenticatorByID`.